src/shared/dm-util.c

```diff
@@ -5,3 +5,43 @@
 #include "dm-util.h"
 #include "fd-util.h"
 #include "string-util.h"
+
+int dm_deferred_remove_cancel(const char *name) {
+        _cleanup_close_ int fd = -1;
+        size_t dm_size = sizeof(struct dm_ioctl) + sizeof(struct dm_target_msg) + STRLEN("@cancel_deferred_remove") + 1;
```

Comment:
hmm, here you calc the size manually even though sizeof() of the structure can do that for you too.

```c
struct message {
        struct…;
        struct…;
        char…;
} message = {
        …
        .dm_ioctl = {
                …
                .data_size = sizeof(struct message);
                …
         }
…
```

The important thing here is that message is now both a struct type here and an instance of that struct. Interestingly in C the struct type is available already for sizeof() in the instance initialization even if it all is just one expression.

Reply:
Neat, didn't know it was possible to use sizeof in the initialization itself. All suggestions applied and pushed.